# Bouyancy calculator. 

From https://study.com/skill/learn/how-to-calculate-the-buoyant-force-of-a-floating-object-explanation.html we have the steps to calculate this and the formula required. 


1. Identify the density of the fluid. 
2. Determine the volume of the fluid displaced by the object.
3. Plud the answers from step 1 and 2 into the equation $Fb = pVg$

Step 1 is easy. Its water. A quick google will tell us the density of water. This will apparently be $p$. Multiple sources from page 1 of google corroborate its $998kg/m^{3}$

Step 2. We need to calculate the volume of the AUV that will be displaced by the water. This is a challenge because of the complex shape. But if we can get something from the CAD designs we might be able to use this? Or we can use the weight and the density of the filament.

https://all3dp.com/2/pla-density-what-s-the-density-of-pla-filament-plastic/ suggests that from most manufacururs the density is $1.24g/cm^{3}$ which is $1240kg/m^{3}$. So all we need is the overall weight of the AUV and we can approximate. This is $V$

Density calculations doesn't factor in the fact the AUV is hollow. Fusion gives the Volume as $7.620E+06mm^{3}$. Factor in the fact parts are printed with 15% infill this is $7.620E+06 \times 0.15 = 1.143E+06mm^{3}$. But we want this in kg/m^{3} so this would be $0.001143$ 

Step 3 is simply calculating the bouyancy using the formula. The $g$ in the formula is the gravitational acceleration (a known constant of 9.8) 

In [26]:
water_density = 1000

#Mass is weight of myself holding the AUV - weight of the AUV. Not the most accurate but its all I have to hand
auv_mass = 100.7 - 96.7
auv_volume = 0.001143
displacement_volume = 0.00762
gravitational_acceleration = 9.8
auv_weight = auv_mass * gravitational_acceleration

bouyancy_force = water_density * displacement_volume * gravitational_acceleration
print("Bouyancy force is: ", bouyancy_force, "N")

Bouyancy force is:  74.676 N


https://courses.lumenlearning.com/suny-osuniversityphysics/chapter/14-4-archimedes-principle-and-buoyancy/

Archimedes principle dictates if an object will sink or float. If the bouyant force is greater than the objects weight it will float. Otherwise the object will sink. 

In [27]:
#We should have both the boyant force and the weight of the auv
print(bouyancy_force)
print(auv_weight)

if bouyancy_force > auv_weight:
    print("The AUV will float")
else:
    print("The AUV will sink")



74.676
39.2
The AUV will float


So we want a bouyant force of 0.1N. 

We should take this off the bouyant force

In [29]:
req_bouyant_force = bouyancy_force - 0.1

print("The required bouyant force is: ", req_bouyant_force, "N")

The required bouyant force is:  74.57600000000001 N


So 74.576 is the target. We should convert this to a mass. Then we can subtract the Mass of the AUV (~4kg).

If weight is mass * grativation acceleration then the mass is the opposite of this.

In [30]:
target_mass = req_bouyant_force / gravitational_acceleration

print("Target mass is: ", target_mass, "kg")

Target mass is:  7.609795918367348 kg


Target mass is 7.61kg (2dp). 

Take of our AUV mass and we have the required ballast. 

In [31]:
ballast_required = target_mass - auv_mass

print("Ballast required is: ", ballast_required, "kg")

Ballast required is:  3.6097959183673476 kg
